cookbooks/aws-parallelcluster-environment/attributes/environment.rb

```diff
@@ -60,3 +60,5 @@
 default['cluster']['internal_initial_shared_dir'] = "#{node['cluster']['base_dir']}/init_shared"
 
 default['cluster']['head_node_private_ip'] = nil
+
+default['cluster']['spack_shared_dir'] = "#{node['cluster']['shared_dir']}/spack"
```

Comment:
Have you tested if using this directory works also when the user logs into the login nodes rather than head node?

Reply:
Good Catch! will move the directory to node['cluster']['base_dir']